### PR TITLE
[feature] Add small CNN for grids 5x5 and up

### PR DESCRIPTION
**Pull request by ervteng**
### Proposed change(s)

Adds a `MATCH3` CNN that can be used for grids down to 5x5. 

### Useful links (Github issues, JIRA tickets, ML-Agents forum threads etc.)

Network architecture used by King in this paper: https://www.researchgate.net/publication/328307928_Human-Like_Playtesting_with_Deep_Learning

### Types of change(s)

- [ ] Bug fix
- [x] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [x] Added tests that prove my fix is effective or that my feature works
- [x] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [x] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents/mlagents/trainers/settings.py [hunk 1]
```diff
@@ -59,6 +59,7 @@ def as_dict(self):
 
 
 class EncoderType(Enum):
+    SMALL = "small"
```
Comment (chriselion) [inline ml-agents/mlagents/trainers/settings.py:62]: Maybe call it "match3"?

**Comment by ervteng** [comment]
> Looks good, any reason not to merge, or you want me to try it on my example first?

Need to add tests for it, but from what I can tell it trains fine and can be merged.

File: ml-agents/mlagents/trainers/settings.py (hunk 1 above)
Comment (ervteng) [reply ml-agents/mlagents/trainers/settings.py]: Renamed to `MATCH3 = "match3"`

**Comment by ervteng** [comment]
Updated the PR - should be ready for production now

**Pull request merged**